### PR TITLE
Copy file_save_upload() to get better error details

### PR DESCRIPTION
**Pull request by amitaibu**

### CODE REVIEW

**Comment by amitaibu** [comment]
@mateu-aguilo-bosch as you can see - few flight related PRs ;)

File: plugins/restful/RestfulFilesUpload.php [hunk 1]
```diff
@@ -117,4 +113,198 @@ public function access() {
...
+   *
+   * @return stdClass
+   *   The saved file object.
+   *
+   * @throws \RestfulBadRequestException
+   * @throws \RestfulServiceUnavailable
+   *
+   * @see file_save_upload()
+   */
+  protected function fileSaveUpload($source) {
```
Comment (e0ipso) [inline plugins/restful/RestfulFilesUpload.php:131]: I know this is an adaptation from core, but this function is way too long and does too many things. I am OK leaving it like this for comparison reasons with `file_save_upload`.

File: restful.module [hunk 2]
```diff
@@ -1025,25 +1025,6 @@ function _restful_invalidate_entity_cache($cid) {
...
-/**
- * Access callback; Determine if user can access file uploading.
- *
- * If "File entity" module exists, determine access by its provided permissions
- * otherwise, check if variable is set to allow anonymous users to upload.
- * Defaults to authenticated user.
- *
- * @return bool
- */
-function restful_file_upload_access() {
```
Comment (e0ipso) [inline restful.module:1038]: Good catch.

**Comment by e0ipso** [comment]
This is ready to go.

**Comment by e0ipso** [comment]
Closed by b4c79a42b47d6e56fd574497ea823d16a0e11c04.

**Pull request closed without merging**